server/src/main/java/org/elasticsearch/index/mapper/ValueFetcherSource.java

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.index.mapper;
+
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.index.query.SearchExecutionContext;
+
+import java.util.List;
+
+/**
+ * Resolves a {@link ValueFetcher} for a particular field.
+ */
+public interface ValueFetcherSource {
+    ValueFetcher preferStored();
```

Comment:
stored here makes me think of stored fields

Reply:
Maybe a bad name. I figured that _source is a stored field. OTOH maybe it's useful to have three layers - `_source`, stored fields, and doc values. Not sure.